packages/data/addon/serializers/metadata/bard.ts

```diff
@@ -278,10 +318,16 @@ export default class BardMetadataSerializer extends NaviMetadataSerializer {
    * @param table - the table to create a dateTime for
    * @param dataSourceName - data source name
    */
-  private createTimeGrainColumnFunction(table: RawTablePayload, dataSourceName: string): ColumnFunctionMetadataModel {
-    const grainIds = table.timeGrains.map(g => this.normalizeTimeGrain(g.name));
-    const grains = grainIds.sort().join(',');
-    const columnFunctionId = `${this.namespace}:timeGrain(table=${table.name};grains=${grains})`;
+  private createTimeGrainColumnFunction(
+    table: RawTablePayload,
+    timeGrainInfo: TableTimeGrainInfo,
+    dataSourceName: string
+  ): ColumnFunctionMetadataModel {
+    const { hasAllGrain } = timeGrainInfo;
+    const grainIds = timeGrainInfo.timeGrains.map(g => g.name);
+
+    const grains = grainIds.join(',');
+    const columnFunctionId = `${this.namespace}:timeGrain(table=${table.name};grains=${grains};hasAllGrain=${hasAllGrain})`;
```

Comment:
What is the case when you need `hasAllGrain=${hasAllGrain}` to distinguish between different column function?

Reply:
Ah good call I'll remove this it shouldn't make a difference